x-pack/plugins/apm/public/components/app/error_group_details/index.tsx

```diff
@@ -39,13 +34,20 @@ import { fromQuery, toQuery } from '../../shared/links/url_helpers';
 
 type ErrorDistributionAPIResponse =
   APIReturnType<'GET /internal/apm/services/{serviceName}/errors/distribution'>;
+type ErrorSamplesAPIResponse =
+  APIReturnType<'GET /internal/apm/services/{serviceName}/errors/{groupId}/samples'>;
 
 const emptyState: ErrorDistributionAPIResponse = {
   currentPeriod: [],
   previousPeriod: [],
   bucketSize: 0,
 };
 
+const emptyErrorSamples: ErrorSamplesAPIResponse = {
+  errorSampleIds: [],
+  occurrencesCount: 0,
+};
+
```

Comment:
I introduced these "initial state" objects years ago before we had optional chaining but today I think it's fairly easy to handle the empty state case:

```ts 
const foo = errorSamplesData?. errorSampleIds.map(...);
```